### PR TITLE
Support custom onCancel function defined in the linked controller.

### PR DESCRIPTION
**Pull request by JacobJust**

### CODE REVIEW

File: src/wizard.js [hunk 1]
```diff
@@ -268,14 +269,19 @@ angular.module('mgo-angular-wizard').directive('wizard', function() {
...
             //cancel is alias for previous.
             this.cancel = function() {
-                //getting index of current step
-                var index = _.indexOf($scope.getEnabledSteps() , $scope.selectedStep);
-                //ensuring you aren't trying to go back from the first step
-                if (index === 0) {
-                    throw new Error("Can't go back. It's already in step 0");
+            	if ($scope.onCancel) {
+                    //onCancel is linked to controller via wizard directive:
+            		$scope.onCancel();
```
Comment (jacobscarter) [inline src/wizard.js]: un-indent 2 spaces

**Comment by jacobscarter** [comment]
@JacobJust can you fix your whitespace and update the README.md so I can merge your PR, thanks for your contributions!

**Comment by JacobJust** [comment]
Fixed the PR comments. Also merged a small branch into this PR, where I send an event when goto another page fails. We want to use this for validation messages in the wizard when next fails.

**Comment by jacobscarter** [comment]
Hey @JacobJust, we just released 0.6.0 which made this branch un-mergable.  Can you please also add a unit test for this? Thanks!

**Comment by jacobscarter** [comment]
@JacobJust I just ficed conflicts, PR merged and slotted for next release

**Pull request merged**